### PR TITLE
Automatically create screens for applist and settings

### PR DESCRIPTION
**Pull request by Riksu9000**
Apps and settings are now stored in a single array (two arrays in total). ~~Adding more entries will automatically create more screens.~~

I'm assuming there's a reason for std::function, because I don't know exactly what it does.

### CODE REVIEW

**Comment by JF002** [comment]
This is a great use for `constexpr` :+1:

**Review by JF002 — APPROVED** [review]

File: src/displayapp/screens/settings/Settings.cpp [hunk 1]
```diff
@@ -1,33 +1,27 @@
 #include "displayapp/screens/settings/Settings.h"
 #include <lvgl/lvgl.h>
 #include <array>
-#include "displayapp/screens/List.h"
-#include "displayapp/Apps.h"
 #include "displayapp/DisplayApp.h"
-#include "displayapp/screens/Symbols.h"
+#include "displayapp/Apps.h"
```
Comment (NeroBurner) [inline src/displayapp/screens/settings/Settings.cpp:5]: please move before `DisplayApp.h` include for alphabetical ordering

File: src/displayapp/screens/ApplicationList.h [hunk 2]
```diff
@@ -21,14 +23,31 @@ namespace Pinetime {
...
         Controllers::Settings& settingsController;
         Pinetime::Controllers::Battery& batteryController;
         Controllers::DateTime& dateTimeController;
 
-        ScreenList<2> screens;
-        std::unique_ptr<Screen> CreateScreen1();
-        std::unique_ptr<Screen> CreateScreen2();
-        // std::unique_ptr<Screen> CreateScreen3();
+        static constexpr int appsPerScreen = 6;
+        static constexpr std::array<Tile::Applications, appsPerScreen * 2> applications {{
```
Comment (NeroBurner) [inline src/displayapp/screens/ApplicationList.h:34]: `nScreens` is explicitly mentioned here as 2. If I want to add another app to the list I need to add the `2` to `3`, which is equivalent to `nScreens`.

We could explicilty set `nScreens` and then calculate the `applications.size()` instead

If I just add an `+1` to add the new application as the only app on the third screen then I get an error at runtime when I swipe to the third screen

File: src/displayapp/screens/settings/Settings.h [hunk 3]
```diff
@@ -17,14 +19,35 @@ namespace Pinetime {
...
 
-        ScreenList<4> screens;
+        Controllers::Settings& settingsController;
 
-        std::unique_ptr<Screen> CreateScreen1();
-        std::unique_ptr<Screen> CreateScreen2();
-        std::unique_ptr<Screen> CreateScreen3();
-        std::unique_ptr<Screen> CreateScreen4();
+        static constexpr int entriesPerScreen = 4;
+        static constexpr std::array<List::Applications, entriesPerScreen * 4> entries {{
```
Comment (NeroBurner) [inline src/displayapp/screens/settings/Settings.h:28]: same as above, `nScreens` is already mentioned to be 4

**Review by NeroBurner — CHANGES_REQUESTED** [review]

File: src/displayapp/screens/ApplicationList.cpp [hunk 4]
```diff
@@ -1,13 +1,23 @@
 #include "displayapp/screens/ApplicationList.h"
 #include <lvgl/lvgl.h>
 #include <array>
```
Comment (NeroBurner) [inline src/displayapp/screens/ApplicationList.cpp:3]: `std::function` and `std::unique_ptr` are used in this file. We should include the corresponding headers [<functional>](https://en.cppreference.com/w/cpp/header/functional) and [<memory>](https://en.cppreference.com/w/cpp/header/memory)

```suggestion
#include <array>
#include <functional>
#include <memory>
```

File: src/displayapp/screens/settings/Settings.cpp [hunk 5]
```diff
@@ -1,33 +1,27 @@
 #include "displayapp/screens/settings/Settings.h"
 #include <lvgl/lvgl.h>
 #include <array>
```
Comment (NeroBurner) [inline src/displayapp/screens/settings/Settings.cpp:3]: `std::function` and `std::unique_ptr` are used in this file. We should include the corresponding headers [<functional>](https://en.cppreference.com/w/cpp/header/functional) and [<memory>](https://en.cppreference.com/w/cpp/header/memory)

```suggestion
#include <array>
#include <functional>
#include <memory>
```

**Review by NeroBurner — CHANGES_REQUESTED** [review]

**Review by NeroBurner — APPROVED** [review]
LGTM

**Pull request merged**